### DISCUSSION TITLE
Running Tests By Group

### DISCUSSION THREAD

**Discussion by TroyRudolph** [Codeception/Codeception#6171] [Ideas]
Hello.  The @group annotation permits us to all tests in a group.  Specifying multiple groups in the "-g" argument ORs the groups together.  Thus, specifying two groups causes all tests in the two groups to be executed. 

I would like to be able to AND the groups together.  For example, we will want to run all tests that are in group "pre_deployment_regression" and in group "student_transcript". 

One way to do this would be to implement a "-G" option that would AND the groups together instead or ORing them together.  (Note the upper case "G".) 

Do you think this could be implemented?

Thank you.

**Comment by DavertMik** [comment]
Probably this can be done in this syntax... 

```
-g groupname1,groupname2
```

Ok, i' not sure If I implement this feature soon. I need to check if PHPUnit internals allow to do this.
If you have time, you can implement it by yourself and send a PR

**Comment by TroyRudolph** [comment]
My thinking was that -g could OR the list and that -G could AND the list.  That way it would be flexible. I hadn't thought about PHPUnit, though I should have.  I'll see what I can find out and get back to you.  Thank you.

**Comment by zbateson** [comment]
Not sure I understand the difference in the explanation between "AND" and "OR".

`Specifying multiple groups in the "-g" argument ORs the groups together. Thus, specifying two groups causes all tests in the two groups to be executed.`

And:

`I would like to be able to AND the groups together. For example, we will want to run all tests that are in group "pre_deployment_regression" and in group "student_transcript".`

Aren't those the same things?  Both will run all tests in the groups - I must be missing something!

**Comment by reinholdfuereder** [comment]
@zbateson was reading very, very thoroughly ;-) => maybe it should be:

> Specifying multiple groups in the "-g" argument ORs the groups together. Thus, specifying two groups causes all tests in **[either (i.e. also just one)]** of the two groups to be executed.
